File: Octave/newtonsMethod.m

1;

# enter the function here
function y = f(x)
    y = 2*x^2-3*x;
endfunction

# enter the derivative here
function y = fp(x)
    y = 4*x-3;
endfunction

# enter the initial guess here

a = .25;

# define the tolerance
tol = 10^(-12);

# define the maximum number of iterations
N = 500;

count = 0;

while abs(f(a))>=tol && count < N
    diff = f(a)/fp(a);
    a = a- diff;
    count = count +1;
endwhile

if count == N
    fprintf("Maximum number of iterations reached\n")
else
    fprintf("The root is approximately: %.5f\n",a)
endif

File: Python/root-approximation/newtons_method.py
import function

# enter initial guess
x0 = 1.5

# enter tolerance
tol = 0.00001

# enter maximum number of iterations
N = 100

# do not change anything beneath this line
n = 0

while n<N:
    n = n+1
    x = x0-function.f(x0)/function.f_prime(x0)
    if abs(x-x0)<tol:
        print(f'f({x})=0')
        print(f'iteration count: {n}')
        break
    x0 = x
if n==N:
    print(f'max iteration count reached.')
